src/frontend/src/lib/utils/device.utils.ts

```diff
@@ -0,0 +1,16 @@
+import { nonNullish } from '@dfinity/utils';
+
+interface UserAgentData {
```

Comment:
Is there a typedef (`@types/...`) available for this? If not, should we declare the type globally in a `.d.ts` file in case it ever become available in NodeJS?

Like declaring it for the `navigator` this way you do not have to explicitely declare the types when you use it.

Strictly a question, not necessarely a call to action.